### PR TITLE
TST: remove workarounds now that contourpy has unpinned nightlies

### PR DESCRIPTION
**Pull request by neutrinoceros**
### Description
This reverts ~#15290~ (and parts of #15363) following https://github.com/contourpy/contourpy/issues/362

EDIT: xref https://github.com/matplotlib/matplotlib/issues/26847#issuecomment-1968516509

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by neutrinoceros** [comment]
hum, I must be doing something wrong: I keep breaking the workflows this is supposed to simplify. I'm wondering if I should just leave it be (don't fix what ain't broke).

**Comment by pllim** [comment]
I think you also have to pull in contourpy dev wheel? I don't see that in tox.ini

**Comment by pllim** [comment]
It annoys me that we have to pull in contourpy dev for matplotlib but I guess that is better than 2 separate jobs.

Anyways, if you have moved on to something else, I can take over later when I have time but maybe I will wait for #15603 to go in first.

**Comment by neutrinoceros** [comment]
Please feel free to take the wheel here ! but we can reassess after https://github.com/astropy/astropy/pull/15603 goes in too. In any case this doesn't seem too urgent :)

**Comment by pllim** [comment]
> take the wheel

I see what you did there! 👏

File: tox.ini [hunk 1]
```diff
@@ -118,6 +116,8 @@ install_command =
     devdeps: python -I -m pip install -v --pre
 
 commands =
+    mpldev: pip install -U --pre -i https://pypi.anaconda.org/scientific-python-nightly-wheels/simple matplotlib
```
Comment (matthewfeickert) [inline tox.ini:119]: :wave: As noted in the CI logs (https://github.com/astropy/astropy/actions/runs/8078075132/job/22069637834?pr=16122#step:10:33) as there are additional dependencies that don't live on the index you'll need to include the PyPI as the `--extra-index-url`. c.f. the [example in the `scientific-python/upload-nightly-action` README](https://github.com/scientific-python/upload-nightly-action/blob/50053b39cfd4eda8a4046cab060998ea1f8b839e/README.md#using-nightly-builds-in-ci)


```suggestion
    mpldev: pip install -U --pre --index-url https://pypi.anaconda.org/scientific-python-nightly-wheels/simple --extra-index-url https://pypi.org/simple matplotlib
```

**Review by matthewfeickert — CHANGES_REQUESTED** [review]

File: tox.ini (hunk 1 above)
Comment (neutrinoceros) [reply tox.ini]: Oh, that's it ! I got confused while cherry-picking parts to revert. `PIP_EXTRA_INDEX_URL` is already set on line 25, but I was indeed effectively ignoring it here. Thanks !

Comment (matthewfeickert) [reply tox.ini]: I'd still in general recommend switching the indexes (to `--index-url https://pypi.anaconda.org/scientific-python-nightly-wheels/simple --extra-index-url https://pypi.org/simple`) to get in the habit of helping guard against what happened to [PyTorch in 2022](https://pytorch.org/blog/compromised-nightly-dependency/). But in this particular case it doesn't matter much, given that the name of the packges exist already on both indexes and you are trusting `scientific-python` to not do anything shady on our index.

Comment (neutrinoceros) [reply tox.ini]: wow, thanks for sharing ! Happy to update this patch if @pllim feels we should

**Review by pllim — COMMENTED** [review]
~Does not hurt to have the extra index guard as suggested in this PR.~ (Changed my mind, see https://github.com/astropy/astropy/pull/16122#discussion_r1518362371)

But we agreed that this is not urgent, so I can take over when the Python 3.10 PR is merged. Thanks, all!

**Comment by pllim** [comment]
`image-mpldev` isn't right, I will have to dig...

```
Numpy: 1.26.4
Scipy: 1.12.0
Matplotlib: 3.8.3
```

File: tox.ini (hunk 1 above)
Comment (pllim) [reply tox.ini]: Revisiting this comment now... The line in the suggestion no longer exists. In tox, we define `PIP_EXTRA_INDEX_URL` in `setenv`, following by `matplotlib>=0.0.dev0` in `deps`. Not sure if I want to mess with `--index-url` now especially that this is not a concern right now. Thanks anyway!

**Comment by pllim** [comment]
To answer my own question, no we do not need to explicitly pull in contourpy-dev because mpl-dev magically does that.

**Comment by pllim** [comment]
I also want to run the cron jobs, especially the "no scipy" one, just in case. When I am happy with this, I will approve and merge.

Thanks!

**Review by pllim — APPROVED** [review]
Logs look fine. Linkheck failure unrelated. Exotic archs shouldn't matter here, so I cancelled them because they take too long to run.

Thanks again!

**Comment by neutrinoceros** [comment]
Thank you for finishing this !

**Pull request merged**